test/Java10andUp/src/org/openj9/test/java/lang/management/ThreadMXBean/APITestJava10.java

```diff
@@ -0,0 +1,252 @@
+		Thread currentThread = Thread.currentThread();
+		long myId = currentThread.getId();
+		myThreads.add(currentThread);
+		long ids[] = new long[MAXDEPTH];
+		HashMap<Long, Integer> expectedDepths = new HashMap<>();
+		for (int requestedDepth = 0; requestedDepth < MAXDEPTH; requestedDepth++) {
+			Thread t = new TestThread(requestedDepth);
+			myThreads.add(t);
+			long id = t.getId();
+			logger.debug("start "+id);
+			t.start();
+			while (t.getState() != Thread.State.WAITING) {
+				try {
+					Thread.sleep(1);
+				} catch (InterruptedException e) {
+					// ignore
+				}
+			}
+			ids[requestedDepth] = id;
+			/*
+			 * every thread will have one stack frame for the run() method,
+			 * plus the specified number of extra frames.
+			 */
+			expectedDepths.put(id, requestedDepth + 1);
+		}
+
+		for (int maxDepth = 0; maxDepth < MAXDEPTH; maxDepth++) {
+			logger.debug("maxDepth = "+maxDepth);
+			for (ThreadInfo ti: myBean.dumpAllThreads(false, false, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				Integer requestedDepthTemp = expectedDepths.get(id);
+				if (Objects.isNull(requestedDepthTemp)) {
+					/* non-test thread */
+					continue;
+				}
+				int requestedDepth = requestedDepthTemp;
+				if (id != myId) {
+					checkDepth(requestedDepth, maxDepth, trace, "dumpAllThreads");
+				} else {
+					Assert.assertTrue(trace.length <= maxDepth, "dumpAllThreads: Wrong stack size when maxDepth = "+maxDepth);
+				}
+			}
+			for (ThreadInfo ti: myBean.getThreadInfo(ids, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				Integer requestedDepthTemp = expectedDepths.get(id);
+				if (null == requestedDepthTemp) {
+					/* non-test thread */
+					continue;
+				}
+				int requestedDepth = requestedDepthTemp;
+				if (id != myId) {
+					checkDepth(requestedDepth, maxDepth, trace, "getThreadInfo");
+				} else {
+					Assert.assertTrue(trace.length <= maxDepth, "getThreadInfo: Wrong stack size when maxDepth = "+maxDepth);
+				}
+			}
+		}
+		synchronized (waitObject) {
+			waitObject.notifyAll();
+		}
+		for (Thread t: myThreads) {
+			if (Thread.currentThread().equals(t)) {
+				continue;
+			}
+			logger.debug("wait for "+t.getId());
+			t.join(10000);
+		}
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testDumpLimitsWithMethodHandles() {
+		final int REQUESTED_DEPTH = 5;
+		Thread t = new MethodHandleTestThread(REQUESTED_DEPTH - 1); // Subtract 1 for the run() method.
+		checkThreadInfo(REQUESTED_DEPTH, t);
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testDumpLimitsWithLambdas() {
+		final int REQUESTED_DEPTH = 5;
+		Thread t = new LambdaTestThread(REQUESTED_DEPTH - 2); // Subtract 1 for the run() method and 1 for the lambda.
+		checkThreadInfo(REQUESTED_DEPTH, t);
+	}
+
+	private void checkThreadInfo(int requestedDepth, Thread t) {
+		t.start();
+		final long ids[] = {t.getId()};
```

Comment:
Not that it matters, but not sure why an array is needed for this single element.

Reply:
Ah, I see now.